runtime/near-vm-runner/src/wasmtime_runner.rs

```diff
@@ -442,16 +450,30 @@ impl crate::PreparedContract for VMResult<PreparedContract> {
             }
         };
 
-        let memory_copy = memory.0;
+        let engine = module.engine();
+
         let config = Arc::clone(&result_state.config);
-        let mut logic = VMLogic::new(ext, context, fees_config, result_state, &mut memory);
-        let engine = store.engine();
+
+        let ctx = Ctx::default();
+
+        let mut store = Store::<Ctx>::new(engine, ctx);
```

Comment:
FWIW I think the code would be ever-so-slightly easier to reason about if `ctx.caller` got cloned here and passed into `WasmtimeMemory::new` as an explicit argument vs. it pulling an arc out from `Store`.